sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -517,29 +517,39 @@ case class GetInternalRowField(child: Expression, ordinal: Int, dataType: DataTy
   }
 }
 
-/** Serializes an input object using Kryo serializer. */
-case class SerializeWithKryo(child: Expression) extends UnaryExpression {
+/**
+ * Serializes an input object using a generic serializer (Kryo or Java).
+ * @param kryo if true, use Kryo. Otherwise, use Java.
+ */
+case class EncodeUsingSerializer(child: Expression, kryo: Boolean) extends UnaryExpression {
```

Comment:
It doesn't have to be this PR, but we should consider just passing the `Serializer` in as a literal since we already have a well defined interface.  This way we could use user defined serializers too.